dlt/common/destination/reference.py

```diff
@@ -265,7 +276,23 @@ def _verify_schema(self) -> None:
                     logger.warning(f"A column {column_name} in table {table_name} in schema {self.schema.name} is incomplete. It was not bound to the data during normalizations stage and its data type is unknown. Did you add this column manually in code ie. as a merge key?")
 
 
-class WithStagingDataset:
+class JobClientMetadataStorage(ABC):
+
+    @abstractmethod
+    def get_stored_schema(self) -> Optional[StorageSchemaInfo]:
+        """Retrieves newest schema from destination storage"""
+        pass
+
+    @abstractmethod
+    def get_stored_schema_by_hash(self, version_hash: str) -> StorageSchemaInfo:
+        pass
+
+    @abstractmethod
+    def get_stored_state(self, state_table: str, pipeline_name: str) -> Optional[str]:
+        """Loads compressed state from destination storage"""
+        pass
+
+class WithStagingDataset(ABC):
```

Comment:
I think ABC is needed as base class, otherwise the abstract method decorator will not fail when not implemented